samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs

```diff
@@ -0,0 +1,130 @@
+            };
+
+            // Add the choices for the prompt.
+            options.Choices.Add(new Choice() { Value = QuickRepliesOption, Action = new CardAction() { Title = QuickRepliesOption, Type = ActionTypes.PostBack, Value = QuickRepliesOption } });
+            options.Choices.Add(new Choice() { Value = FacebookPageNameOption, Action = new CardAction() { Title = FacebookPageNameOption, Type = ActionTypes.PostBack, Value = FacebookPageNameOption } });
+            options.Choices.Add(new Choice() { Value = PostBackOption, Action = new CardAction() { Title = PostBackOption, Type = ActionTypes.PostBack, Value = PostBackOption } });
+
+            return await stepContext.PromptAsync(nameof(ChoicePrompt), options, cancellationToken);
+        }
+
+        private async Task<DialogTurnResult> ShowResultStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            Logger.LogInformation("MainDialog.ShowResultStep");
+            var turnContext = stepContext.Context;
+
+            // Initially the bot offers to showcase 3 Facebook features: Quick replies, PostBack and getting the Facebook Page Name.
+            switch (turnContext.Activity.Text)
+            {
+                // Here we showcase how to obtain the Facebook page name.
+                // This can be useful for the Facebook multi-page support provided by the Bot Framework.
+                // The Facebook page name from which the message comes from is in turnContext.Activity.Recipient.Name.
+                case FacebookPageNameOption:
+                    {
+                        var reply = turnContext.Activity.CreateReply($"This message comes from the following Facebook Page: {turnContext.Activity.Recipient.Name}");
+                        await turnContext.SendActivityAsync(reply);
+                        await ShowChoices(stepContext, cancellationToken);
+
+                        break;
+                    }
+
+                // Here we send a HeroCard with 2 options that will trigger a Facebook PostBack.
+                case PostBackOption:
+                    {
+                        var card = new HeroCard
+                        {
+                            Text = "Is 42 the answer to the ultimate question of Life, the Universe, and Everything?",
+                            Buttons = new List<CardAction>
+                                    {
+                                        new CardAction() { Title = "Yes", Type = ActionTypes.PostBack, Value = "Yes" },
+                                        new CardAction() { Title = "No", Type = ActionTypes.PostBack, Value = "No" },
+                                    },
+                        };
+
+                        var reply = turnContext.Activity.CreateReply();
+                        reply.Attachments = new List<Attachment> { card.ToAttachment() };
+                        await turnContext.SendActivityAsync(reply);
```

Comment:
add cancellationToken to SendActivityAsync() call.

Reply:
addressed